src/odemis/gui/comp/miccanvas.py

```diff
@@ -1165,6 +1166,110 @@ def update_drawing(self):
 
         if self.view:
             self.update_thumbnail()
+            
+class NavigableBarPlotCanvas(BarPlotCanvas):
+    
+    def __init__(self, *args, **kwargs):
+        super(NavigableBarPlotCanvas, self).__init__(*args, **kwargs)
+        self.abilities |= {CAN_DRAG}
+        
+        # When resizing, margin to put around the current size
+        # TODO: Maybe make the margin related to the canvas size?
+        self.default_margin = 512
+        self.margins = (self.default_margin, self.default_margin)
+
+        # the position the view is asking to the next buffer recomputation
+        # in buffer-coordinates: = 1px at scale = 1
+        self._data_buffer = None
+        self.window_x = None
+        self.window_y = None
+        self.display_xrange = None
+        self.display_yrange = None
+        self.data_xrange = None
+        self.data_yrange = None
+
+    def set_1d_data(self, xs, ys, unit_x=None, unit_y=None, range_x=None, range_y=None):
+        if len(xs) != len(ys):
+            msg = "X and Y list are of unequal length. X: %s, Y: %s, Xs: %s..."
+            raise ValueError(msg % (len(xs), len(ys), str(xs)[:30]))
+        self._data_buffer = (numpy.array(xs), numpy.array(ys))
+        self.unit_x = unit_x
+        self.unit_y = unit_y
+        self.data_xrange = range_x
+        self.data_yrange = range_y
+        self.display_xrange = range_x
+        self.display_yrange = range_y
+        self.set_ranges(range_x, range_y)
+        
+    def fit_view_to_content(self, recenter=None):
+        # note - need to do this via an event to the viewport. Otherwise it doesn't
+        # set the axes properly
+        self.reset_ranges()
+
+    def reset_ranges(self):
+        if self._data_buffer is None:
+            return
+
+        self.set_ranges(self.data_xrange, self.data_yrange)
+
+    def set_ranges(self, x_range, y_range):
+
+        if self._data_buffer is None:
+            return
+
+        (xst , yst) = self._data_buffer
+
+        # Deal with x
```

Comment:
Please explain more the goal of this code. It's to cache just the data needed to be displayed in the selected range. Something like this, right?

Reply:
You still need to add a comment here explaining what's happening here. "Deal with x" is not enough!